// clairity-client/src/components/layout.jsx
// Layout Grid

var React = require('react');
var objectAssign = require('object-assign');


var Layout = React.createClass({
  propTypes: {
    
    // colNum, ex 12
    cols: React.PropTypes.number,
    
    // {name: [colWidth, colWidth, colWidth], name: [colWidth, colWidth, colWidth]}, ex {lg: [7, 3, 2], md: [4, 4, 4], xs: [12, 12, 12]}
    widths: React.PropTypes.object,
    
    // {name: pxValue, name: pxValue}, ex {lg: 1200, md: 990, sm: 750, xs: 480, xxs: 380}
    breakpoints: React.PropTypes.object,
    
    // {name: [colN, colN, colN], name: [colN, colN, colN]}, ex {md: [2, 1, 3], sm: [1, 2, 3]}
    order: React.PropTypes.object
  
  },
  
  getChildWidth: function() {
    var contWidth = 1;
    var col = this.props.cols || 12 ;
    var widths = this.props.widths || this.props.children.length;
    
    return [contWidth, col, widths.lg];
  },
  
  style: {
    border: "1px solid #ff0000",
    overflow: "auto"
  },
  
  render() {
    
    return (
      <div style={this.style}>
        {this.props.children}
      </div>
    );
  }
});


export default Layout;
