// clairity-client/src/components/table.jsx
import React from 'react'
import Settings from './settings'
import mui from 'material-ui'
import FixedDataTable, {Table, Column, ColumnGroup as Group} from 'fixed-data-table'

let DataTable = React.createClass({

  propTypes: {
    colNames: React.PropTypes.array,
    data: React.PropTypes.array,
    colWidths : React.PropTypes.array,
    maxWidth: React.PropTypes.number
  },

  rowGetter: function(rowIndex) {
      return this.props.data[rowIndex];
  },

  getWidth: function() {
    let width = window.innerWidth - Settings.leftNavWidth - (2 * Settings.contentPadding) - Settings.widthBuffer;
    return width;
  },

  getColWidth: function(i) {
    if(this.props.colWidths) {
      return (this.state.width * (this.props.colWidths[i] / this.props.maxWidth));
    } else {
      return (this.state.width / this.props.colNames.length);
    }
  },

  handleResize: function() {
    this.setState({width: this.getWidth()});
  },

  componentDidMount: function() {
    window.addEventListener('resize', this.handleResize);
  },

  getInitialState: function() {
    return { width: this.getWidth() };
  },

  style: function() {
    return {
      width: this.state.width + "px",
      margin: "20px 0"
    };
  },

  onRowSelect: function(e, index) {
      var value = dfdfdocument.getElementsByClassName('public_fixedDataTableCell_cellContent')[index];
      value.classList.toggle('active-row');
      console.log(value);
  },

  render: function() {
    let columns =
      <Group fixed={true}>
        {
          this.props.colNames.map((name, i) =>
            <Column
              label={name}
              dataKey={i}
              width={this.getColWidth(i)} />
          , this)
        }
      </Group>;

    return (
      <div style={this.style()} className="table">
        <Table
          rowHeight={50}
          onRowClick={this.onRowSelect}
          rowGetter={this.rowGetter}
          rowsCount={this.props.data.length}
          width={this.state.width}
          height={400}
          headerHeight={50}>
            {columns}
        </Table>
      </div>
    );
  }
});

export default DataTable;
