// clairity-client/src/components/table.jsx
import React from 'react'
import Settings from './settings'
import mui from 'material-ui'

// Polyfill not working with import
Object.assign = require('object-assign');
var FixedDataTable = require('fixed-data-table')


var Table = FixedDataTable.Table;
var Column = FixedDataTable.Column;
var Group = FixedDataTable.ColumnGroup

var DataTable = React.createClass ({

  propTypes: {
    colNames: React.PropTypes.array,
    data: React.PropTypes.array
  },

  rowGetter: function(rowIndex) {
      return this.props.data[rowIndex];
  },

  getWidth: function() {
    var width = window.innerWidth - Settings.leftNavWidth - (2 * Settings.contentPadding);
    return width;
  },

  handleResize: function() {
    this.setState({width: this.getWidth()});
  },

  componentDidMount: function() {
    window.addEventListener('resize', this.handleResize);
  },

  getInitialState: function() {
    return { width: this.getWidth() };
  },

  style: function() {
    return {
      width: this.state.width + "px",
      margin: "20px 0"
    };
  },

  render: function() {

    console.log(this.state.width);

    var columns =
      <Group fixed={true}>
          {this.props.colNames.map(function(name, i) {
             return <Column label={name} dataKey= {i} width={ this.state.width / this.props.colNames.length }/>;
          }, this)}
      </Group>;

    console.log(this.props.data.length);

    return (
      <div style={this.style()} className="table">
          <Table
          rowHeight={50}
          rowGetter={this.rowGetter}
          rowsCount={this.props.data.length}
          width={this.state.width}
          height={400}
          headerHeight={50}>
            {columns}
        </Table>
      </div>
    );
  }
});

export default DataTable;
