// clairity-client/src/components/pages/aging.jsx
import React from 'react'
import Layout from  '../layout'
import Footer from  '../footer'
import TopNav from '../topnav'
import LeftNav from '../leftnav'
import Table from '../table'
import Content from '../content'
import Filters from '../filters'

// TODO: Get view listening to the store correctly. (Generic solution?)
import Store from '../../store'

// TODO: Combine all the disparate logic into a single 'feature'
import {getAgingReports} from '../../actions/aging'
import {getActive, getInactive, getAll} from '../../queries/aging'

// Filter components

import {
  RadioButton,
  RadioButtonGroup,
  Checkbox,
  FlatButton,
  RaisedButton,
  FloatingActionButton,
  IconButton,
  Toggle,
  Slider,
  DropDownMenu,
  DatePicker,
  TextField
} from 'material-ui'

// Table Data
let colNames = [
  'Customer',
  'Status',
  'Balance',
  '0 - 30',
  '31 - 60',
  '61 - 90',
  '91+',
  'Send Late Notice',
  'Last Weekly Notice',
  'Last Monthly Notice',
  'Agent',
  'State'
];

let rows = 'a b c d e f g h i j k l m n o p q r s t u v w x y z'
  .split(' ')
  .map(c => colNames.map(n => `${c} - ${n}`));

export default class Aging extends React.Component {
  constructor() {
    super();
    this.state = {
      colNames: colNames,
      rows: rows
    };
  }
  componentDidMount() {
    getAgingReports();
  }
  render() {
    return (
      <div>
        <TopNav />
        <div className="main">
          <Layout>
            <LeftNav />
            <Content>
              <Layout widths={{lg:[12],md:[12],sm:[12],xs:[12]}}>
                <div>
                  <div className="section-header">
                    <h1>Aging Reports</h1>
                  </div>
                  <Filters data={rows}>
                    <RadioButtonGroup name="status" defaultSelected="both">
                      <RadioButton value="active" label="Acitve" />
                      <RadioButton value="inactive" label="Inactive"  defaultChecked={true} />
                      <RadioButton value="both" label="Both" />
                    </RadioButtonGroup>
                    <Checkbox  name="checkboxName1" value="checkboxValue1"  label="Hide $0 Credit Balances" />
                    <DatePicker hintText="As of" mode="landscape" />
                  </Filters>
                  <Table data={this.state.rows} colNames={this.state.colNames} />
                </div>
              </Layout>
            </Content>
          </Layout>
        </div>
        <Footer />
      </div>
    );
  }
};
