// clairity-client/src/components/pages/aging.jsx
import React from 'react'
import Layout from  '../layout'
import Footer from  '../footer'
import TopNav from '../topnav'
import LeftNav from '../leftnav'
import Table from '../table'
import Content from '../content'
import Filters from '../filters'

import {fromJS, List} from 'immutable'

// TODO: Get view listening to the store correctly. (Generic solution?)
import Store from '../../store'

// TODO: Combine all the disparate logic into a single 'feature'
import {getAgingReports} from '../../actions/aging'
import {getActive, getInactive, getAll} from '../../queries/aging'

// Filter components

import {
  RadioButton,
  RadioButtonGroup,
  Checkbox,
  FlatButton,
  RaisedButton,
  FloatingActionButton,
  IconButton,
  Toggle,
  Slider,
  DropDownMenu,
  DatePicker,
  TextField
} from 'material-ui'

// Table Data
let colNames = [
  'Customer',
  'Status',
  'Balance',
  '0 - 30',
  '31 - 60',
  '61 - 90',
  '91+',
  'Send Late Notice',
  'Last Weekly Notice',
  'Last Monthly Notice',
  'Agent',
  'State'
];

let rows = 'a b c d e f g h i j k l m n o p q r s t u v w x y z'
  .split(' ')
  .map(c => colNames.map(n => `${c} - ${n}`));

export default class Aging extends React.Component {
  constructor() {
    super();
    this.state = {
      colNames: colNames,
      rows: fromJS(rows)
    };
    this.getStateFromStore = this.getStateFromStore.bind(this);
  }
  getStateFromStore() {
    this.setState({rows: getAll(false)});
  }
  componentWillMount() {
    getAgingReports();
  }
  componentDidMount() {
    this.getStateFromStore();
    Store.on('update', this.getStateFromStore);
  }
  componentWillDismount() {
    // TODO: Add a Store.off method
    //Store.off('update', this.getStateFromStore);
  }
  computeRows() {
    return this.state.rows.toList().toJS()
      .map(r => this.rowToArray(r));
  }
  rowToArray(r) {
    return [
      r.name,
      r.active,
      r.b, r.b1, r.b2, r.b3, r.b4,
      r.button_disabled,
      r.weekly_late_notice_sent,
      r.late_notice_sent,
      r.agent
    ];
  }
  render() {
    let rowsData = this.computeRows();
    return (
      <div>
        <TopNav />
        <div className="main">
          <Layout>
            <LeftNav />
            <Content>
              <Layout widths={{lg: [12], md: [12], sm: [12], xs: [12]}}>
                <div>
                  <div className="section-header">
                    <h1>Aging Reports</h1>
                  </div>
                  <Filters data={rows}>
                    <RadioButtonGroup name="status" defaultSelected="both">
                      <RadioButton value="active" label="Acitve" />
                      <RadioButton value="inactive" label="Inactive"  defaultChecked={true} />
                      <RadioButton value="both" label="Both" />
                    </RadioButtonGroup>
                    <Checkbox  name="checkboxName1" value="checkboxValue1"  label="Hide $0 Credit Balances" />
                    <DatePicker hintText="As of" mode="landscape" />
                  </Filters>
                  <Table
                    data={rowsData}
                    colNames={this.state.colNames}
                    colWidths={[4,1,1,1,1,1,1,2,2,2,2]}
                    maxWidth={18} />
                </div>
              </Layout>
            </Content>
          </Layout>
        </div>
        <Footer />
      </div>
    );
  }
};
