// clairity-client/src/components/pages/editCustomer.jsx
import React from 'react'
import Layout from  '../layout'
import Footer from  '../footer'
import TopNav from '../topnav'
import LeftNav from '../leftnav'
import Content from '../content'
import {
  RadioButtonGroup,
  RadioButton,
  Checkbox,
  FlatButton,
  RaisedButton,
  FloatingActionButton,
  IconButton,
  Toggle,
  Slider,
  DropDownMenu,
  DatePicker,
  TextField,
  Paper
} from 'material-ui'

import {List, fromJS} from 'immutable'

// Dropdown Menu Data

var typeItems = [
  { payload: '1', text: 'Business' },
  { payload: '2', text: 'Residential' },
  { payload: '3', text: 'Telenational - Business' },
  { payload: '4', text: 'Telenational - Residential' }
];

var locationItems = [
  { payload: '1', text: '1845 Oil Field Services' },
  { payload: '2', text: '150 Capital Ave, Atlanta GA' },
];

var typeItems = [
   { payload: '1', text: 'Type' },
   { payload: '2', text: 'Month-to-Month' },
   { payload: '3', text: 'One-time Event' },
   { payload: '4', text: 'Standard - Downgrade' },
   { payload: '5', text: 'Standard - New Install' },
   { payload: '6', text: 'Standard - Renewal' },
   { payload: '7', text: 'Standard - Upgrade' }
];

var salesItems = [
   { payload: '1', text: 'Sales Agent' },
   { payload: '2', text: 'Demo Employee' },
   { payload: '3', text: 'Andrew Solomon' },
   { payload: '4', text: 'Brad Hackett' },
   { payload: '5', text: 'Chriss Meadows' }
];

var serviceItems = [
   { payload: '1', text: 'Choose a service ...' },
   { payload: '2', text: 'Demo Employee' },
   { payload: '3', text: 'Andrew Solomon' },
   { payload: '4', text: 'Brad Hackett' },
   { payload: '5', text: 'Chriss Meadows' }
];

// React Render
export default class CreateContract extends React.Component {
  constructor() {
    super();
    this.state = {
      services: new List()
    };
  }
  addService() {
    this.setState({services: this.state.services.push({})});
  }
  removeService(n) {
    this.setState({services: this.state.services.delete(n)});
  }
  render() {
    return (
      <div>
        <TopNav />
        <Layout type="main">
          <LeftNav />
          <Content>
            <Layout>
              <div className="section-header">
                <h1>Create Contract</h1>
              </div>
              <Paper zDepth={1} rounded={true}>
                <Layout widths={{ lg: [6, 6], md: [12, 12], sm: [12, 12], xs: [12, 12]}} cPadding={'0 20px 20px 20px'}>
                  <div>
                    <h4>Customer(Billing) Details</h4>
                    <DropDownMenu menuItems={typeItems} />
                    <Layout widths={{ lg: [12,6,6,12,6,6,4,4,4], md: [12,6,6,12,6,6,4,4,4], sm: [12,6,6,12,6,6,4,4,4], xs: [12,6,6,12,6,6,4,4,4]}}>
                      <TextField
                        defaultValue=""
                        floatingLabelText="Name" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Street 1" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Street 2" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="City" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="State" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Zip Code" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Reference #" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Attn" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Legacy Account #" />
                    </Layout>
                    <Layout widths={{ lg: [6,6,6,6,6,6,6,6], md: [6,6,6,6,6,6,6,6], sm: [6,6,6,6,6,6,6,6], xs: [6,6,6,6,6,6,6,6]}} pPadding={"20px 0"}>
                      <Checkbox
                        name="taxExempt"
                        value="taxExempt"
                        label="Tax Exempt" />
                      <Checkbox
                        name="summaryBilling"
                        value="summarryBilling"
                        label="Summary Billing" />
                      <Checkbox
                        name="showInternationalCalls"
                        value="showInternationalCalls"
                        label="Show International Calls" />
                      <Checkbox
                        name="showLongDistanceCalls"
                        value="showLongDistanceCalls"
                        label="Show long Distance Calls" />
                      <Checkbox
                        name="emailInvoice"
                        value="emailInvoice"
                        label="Email Invoice" />
                      <Checkbox
                        name="invoiceWeekly"
                        value="invoiceWeekly"
                        label="Invoice Weekly" />
                      <Checkbox
                        name="vip"
                        value="vip"
                        label="VIP" />
                      <Checkbox
                        name="autoPlay"
                        value="autoPlay"
                        label="Auto Pay" />
                    </Layout>
                  </div>
                  <div>
                    <h4>Service Location</h4>
                    <Layout widths={{ lg: [12,12,12,6,6,12,6,6,4,4,4], md: [12,12,12,12,12,12,12], sm: [12,12,12,12,12,12,12], xs: [12,12,12,12,12,12,12]}}>
                      <TextField
                        defaultValue=""
                        floatingLabelText="Name" />
                      <Checkbox
                        name="taxExempt"
                        value="taxExempt"
                        label="Tax Exempt" />
                      <Checkbox
                        name="taxExempt"
                        value="taxExempt"
                        label="Same as Customer Address" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Street 1" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Street 2" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="City" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="State" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Zip Code" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Reference #" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Legacy Account #" />
                      <TextField
                        defaultValue=""
                        floatingLabelText="Order Index" />
                    </Layout>
                  </div>
                </Layout>
              </Paper>
              <Layout pPadding={"10px 0"}>
                <RaisedButton label="Submit" primary={true} />
              </Layout>
            </Layout>
          </Content>
        </Layout>
        <Footer />
      </div>
    );
  }
}
