// clairity-client/src/components/pages/login.jsx

import React, {PropTypes} from 'react'
import Layout from  '../layout'
import Footer from  '../footer'
import TopNav from '../topnav'
import LeftNav from '../leftnav'
import Content from '../content'
import {
  RadioButtonGroup,
  RadioButton,
  Checkbox,
  FlatButton,
  RaisedButton,
  FloatingActionButton,
  IconButton,
  Toggle,
  Slider,
  DropDownMenu,
  DatePicker,
  TextField,
  Paper
} from 'material-ui'

import controllable from 'react-controllables'

import {tryLogin, loginSuccess} from '../../actions/login'
import {} from '../../queries/auth'

class LoginForm extends React.Component {
  changeUsername(ev) {
    if (!this.props.onUsernameChange) return;
    this.props.onUsernameChange(ev.target.value);
  }
  changePassword(ev) {
    if (!this.props.onPasswordChange) return;
    this.props.onPasswordChange(ev.target.value);
  }
  fireLogin() {
    if (this.props.onLogin) {
      this.props.onLogin(this.props.username, this.props.password);
    }
  }
  render() {
    return (
      <div>
        <Paper style={{position: 'relative'}}>
          <div>
            <TextField
              value={this.props.username}
              onChange={this.changeUsername.bind(this)}
              floatingLabelText="Username" />
          </div>
          <div>
            <TextField
              value={this.props.password}
              floatingLabelText="Password"
              onChange={this.changePassword.bind(this)}
              type="password" />
          </div>
          <div>
          </div>
            <RaisedButton onClick={this.fireLogin.bind(this)}>
              <div style={{height: 36, lineHeight: '36px', color: 'blue !important'}}>Login</div>
            </RaisedButton>
        </Paper>
      </div>
    );
  }
}

LoginForm.defaultProps = { username: '', password: '' }
LoginForm.propTypes = {
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  onLogin: PropTypes.func
}

let Form = controllable(LoginForm, ['username', 'password']);
import {Navigation} from 'react-router'

const Login = React.createClass({
  mixins: [Navigation],
  componentDidMount() {
    loginSuccess.listen(() => this.transitionTo('aging-reports'));
  },
  tryLogin(username, password) {
    tryLogin(username, password);
  },
  render() {
    return (
      <Form defaultUsername="ornemployee@yahoo.com" onLogin={this.tryLogin} />
    );
  }
});

export default Login;
