// clairity-client/src/components/pages/workOrders.jsx

import React, {PropTypes} from 'react'
import Store from '../../store'
import Layout from '../layout'
import Footer from '../footer'
import TopNav from '../topnav'
import LeftNav from '../leftnav'
import Content from '../content'
import {
  RadioButtonGroup,
  RadioButton,
  Checkbox,
  FlatButton,
  RaisedButton,
  FloatingActionButton,
  IconButton,
  Toggle,
  Slider,
  DropDownMenu,
  DatePicker,
  TextField,
  Paper
} from 'material-ui'

import controllable from 'react-controllables'

import {getWorkOrder, updateWorkOrder} from '../../actions/workOrders'
import {getOrder} from '../../queries/workOrder'
import {Navigation} from 'react-router'


let WorkOrders = React.createClass({
  mixins: [Navigation],

  conmponentWillMount() {
    getWorkOrder(this.getParams().id);
  },

  componentDidMount() {
    Store.on('update', () => this.setState( this.getStateFromStore() ));
  },

  getStateFromStore() {
    return { order: queryWorkOrder(this.getParams(id)) };
  },

  tryLogin(username, password) {
    tryLogin(username, password);
  },

  updateAddressField(fieldName) {
    return value => {
      this.setState({order: this.state.order.setIn(['address', fieldName], value)});
    }
  },

  updateOrder() {
    let id = this.getParams().id;
    updateWorkOrder(id, this.state.order);
  },

  render() {
    return (
      <div>This is workorders</div>
      //value={this.state.order.getIn(['address', 'street'])} onChange={this.updateAddressField('street')}
    )
  }
});

export default WorkOrders;
